Increase API rate limit thresholds to 99%

## Summary

The shepherd currently stops at 90% API rate limit threshold, which is too conservative. Increase all rate limit thresholds to 99%.

## Current Behavior

Shepherd exits with error when API usage reaches 90%:
```
[ERROR] API rate limit exceeded (threshold: 90%)
```

## Desired Behavior

Allow operations to continue until 99% rate limit usage.

## Rationale

- 90% threshold is overly cautious and interrupts legitimate work
- Rate limits reset regularly, so hitting them occasionally is acceptable
- Better to complete work and hit the limit than to stop prematurely

## Acceptance Criteria

- [ ] Update all rate limit thresholds from 90% to 99%
- [ ] Verify shepherd continues past previous 90% threshold

--- a/loom-tools/src/loom_tools/shepherd/config.py
+++ b/loom-tools/src/loom_tools/shepherd/config.py
@@ -96,7 +96,7 @@ class ShepherdConfig:
 
     # Rate limiting
     rate_limit_threshold: int = field(
-        default_factory=lambda: _get_env_int("LOOM_RATE_LIMIT_THRESHOLD", 90)
+        default_factory=lambda: _get_env_int("LOOM_RATE_LIMIT_THRESHOLD", 99)
     )
 
     # Worktree marker file name

--- a/loom-tools/tests/shepherd/test_errors.py
+++ b/loom-tools/tests/shepherd/test_errors.py
@@ -75,15 +75,15 @@ class TestRateLimitError:
 
     def test_stores_usage_and_threshold(self) -> None:
         """Error should store usage and threshold values."""
-        err = RateLimitError(95.5, 90.0)
-        assert err.usage_percent == 95.5
-        assert err.threshold == 90.0
+        err = RateLimitError(99.5, 99.0)
+        assert err.usage_percent == 99.5
+        assert err.threshold == 99.0
 
     def test_formats_message(self) -> None:
         """Error message should include usage percentage."""
-        err = RateLimitError(95.5, 90.0)
-        assert "95.5%" in str(err)
-        assert "90.0%" in str(err)
+        err = RateLimitError(99.5, 99.0)
+        assert "99.5%" in str(err)
+        assert "99.0%" in str(err)
 
 
 class TestIssueNotFoundError:

